### ISSUE TITLE
Release 2.5 Introduces potential issue to perception offline simulation

### ISSUE THREAD

**Issue by TuanHo16** [ApolloAuto/apollo#4127]
Hi Team,

Thanks for releasing the latest version of Apollo.

I am able to build the offline perception and run the Lidar visualizer without any problem for v2.5 and the previous version.

However, there is one potential issue with the gcc version for the top-level build that invoked by `apollo.sh`.

When I install the nVidia GPU driver in the docker, I have to roll back the gcc version and re-roll the version of current gcc, similar to what described in the scripts: [https://github.com/ApolloAuto/apollo/blob/master/scripts/install_gcc.sh](url), and
[https://github.com/ApolloAuto/apollo/blob/master/scripts/recover_gcc.sh](url)

The nVidia driver installation went smoothly for v2.5 and previous version. However, in v2.5 this causes a minor gcc error (the final result is still passed) in : ./apollo.sh build_gpu as follows:

```
$ ./apollo.sh build_gpu

Tuan@in_dev_docker:/apollo$ ./apollo.sh build_gpu
System check passed. Build continue ...
[WARNING] ESD CAN library supplied by ESD Electronics does not exist. If you need ESD CAN, please refer to third_party/can_card_library/esd_can/README.md.
[INFO] Start building, please wait ...
INFO: Reading 'startup' options from /apollo/tools/bazel.rc: --batch_cpu_scheduling
[INFO] Building on x86_64...
INFO: Reading 'startup' options from /apollo/tools/bazel.rc: --batch_cpu_scheduling
INFO: Analysed 1480 targets (9 packages loaded).
INFO: Found 1480 targets...
INFO: From Executing genrule //modules/perception/cuda_util:cuda_util_genrule:
-- Configuring done
-- Generating done
-- Build files have been written to: /home/tuan/.cache/bazel/_bazel_tuan/540135163923dd7d5820f3ee4b306b32/execroot/apollo/modules/perception/cuda_util/cmake_build
[ 14%] Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_block_uf.cu.o
[ 28%] [ 42%] Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_undistortion.cu.o
Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_util.cu.o
[ 57%] [ 71%] Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_region_output.cu.o
[ 85%] Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_network.cu.o
Building NVCC (Device) object CMakeFiles/cuda_compile.dir//./cuda_compile_generated_connected_component_gpu.cu.o
Linking CXX shared library libcuda_util.so
[100%] Built target cuda_util
/home/tuan/.cache/bazel/_bazel_tuan/540135163923dd7d5820f3ee4b306b32/execroot/apollo
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
/usr/include/eigen3/Eigen/src/Core/PlainObjectBase.h(127): warning: statement is unreachable

cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
cc1: warning: command line option '-std=c++11' is valid for C++/ObjC++ but not for C [enabled by default]
INFO: From Executing genrule //modules/drivers/pandora:pandora_genrule:
CMake Error at cmake_install.cmake:54 (FILE):
  file INSTALL cannot copy file
  "/tmp/build_isolated/pandora_driver/catkin_generated/installspace/_setup_util.py"
  to "/home/tmp/ros/_setup_util.py".


make: *** [install] Error 1
Command failed, exiting.
CMake Error at cmake_install.cmake:54 (FILE):
  file INSTALL cannot copy file
  "/tmp/build_isolated/pandora_pointcloud/catkin_generated/installspace/_setup_util.py"
  to "/home/tmp/ros/_setup_util.py".


make: *** [install] Error 1
Command failed, exiting.
INFO: Elapsed time: 933.268s, Critical Path: 46.73s
INFO: Build completed successfully, 4811 total actions
============================
[ OK ] Build passed!
[INFO] Took 936 seconds
============================
```
I also clone a new apollo and just compile it for top-level simulation for testing. Without having to run the driver installation in the docker (for offline perception sim), the top-level build went fine without any issue.

My question: have you guys changed the version of gcc between v2.0 and v2.5 and update it according into the following script?

[https://github.com/ApolloAuto/apollo/blob/master/scripts/recover_gcc.sh](url)

Best regards,
Tuan

**Comment by jinghaomiao** [comment]
@mickeyouyou can you please help with it? It looks like it's the pandora driver issue.

**Comment by mickeyouyou** [comment]
OK, we will try build on gpu.

**Comment by mickeyouyou** [comment]
we have fixed in #4252  you can try build on gpu again.
